test/pkg/environment/aws/environment.go

```diff
@@ -90,12 +90,11 @@ func NewEnvironment(t *testing.T) *Environment {
 		IAMAPI:        iam.New(session),
 		FISAPI:        fis.New(session),
 		EKSAPI:        eks.New(session),
-		SQSProvider:   interruption.NewSQSProvider(sqs.New(session)),
+		SQSProvider:   lo.Must(sqs.NewProvider(env.Context, servicesqs.New(session), lo.Must(os.LookupEnv("INTERRUPTION_QUEUE")))),
```

Comment:
I think I might've asked this before, but wouldn't this panic if the queue doesn't exist? 

https://github.com/aws/karpenter/pull/4954/files#diff-89f1d90b65fce5d8e3d30df41441d721cecea16149000f221ceb29db20761354R38-R40

Should we be putting a lo.Must() around a function that we know can fail? If it can't, can we remove the error in the function itself?

Reply:
I think it's reasonable to expect that the queue exists as a pre-requisite for the provider to be created. In this case, every E2E test is assumed to have a queue. Are you thinking that we should only create the provider if we are doing interruption testing?

Reply:
We have customers who aren't using the queue, and Karpenter shouldn't crash if they are not using interruption. We've taken issues on this before.